syntax = "proto3";


package TP2;

// The greeting service definition.
service Central {
  // Sends a greeting
  rpc Register (RegisterRequest) returns (RegisterReply) {}
  rpc Terminate (CentralTerminateRequest) returns (CentralTerminateReply) {}
  rpc Map (MapRequest) returns (MapReply) {}
}

// The request message containing the user's name.
message RegisterRequest {
  string id = 1;
  repeated uint32 keys = 2;
}

// The response message containing the greetings
message RegisterReply {
  int32 result = 1;
}

// The request message containing the user's name.
message MapRequest {
  uint32 key = 1;
}

// The response message containing the greetings
message MapReply {
  string id = 1;
}

// The request message containing the user's name.
message CentralTerminateRequest {
  uint32 key = 1;
  string value = 2;
}

// The response message containing the greetings
message CentralTerminateReply {
  int32 ret = 1;
}
